src/pkgcore/ebuild/pkg_updates.py

```diff
@@ -8,22 +8,19 @@
 from ..log import logger
 from .atom import atom
 
-demand_compile_regexp('valid_updates_re', r'^([1-4])Q-(\d{4})$')
 
-
-def _scan_directory(path):
+def _scan_directory(path, eapi):
     files = []
     for filename in listdir_files(path):
-        match = valid_updates_re.match(filename)
+        match = eapi.options.update_regex.match(filename)
         if match is not None:
-            files.append(((match.group(2), match.group(1)), filename))
+            files.append(filename)
         else:
             logger.error(f'incorrectly named update file: {filename!r}')
-    files.sort(key=itemgetter(0))
-    return [x[1] for x in files]
+    return files
```

Comment:
I don't think sorting should be removed here even though the spec doesn't specify it's required since the order of iteration probably affects pkgcheck output among other things. If EAPI 8 doesn't care what the files are named, then use a basic lexical sort.

Reply:
Done.